### ISSUE TITLE
Component without a template causes an error

### ISSUE THREAD

**Issue by stogle** [ipatalas/ngComponentUtility#16]
A component without a `template` or `templateUrl` defined causes the error `Error refreshing components, check developer console` and the extension stops working.

This also happens if the component is a class that inherits its `template` or `templateUrl` from a base class, rather than defining it directly, e.g.:

```TS
export abstract class BaseComponent implements ng.IComponentOptions {
    constructor(public controller: ng.Injectable<ng.IControllerConstructor>) {
    }

    public template: string = require("./template.html");
}

export class TestComponent extends BaseComponent {
    constructor() {
        super(TestController);
    }
}

export class TestController {
}
```

Note that in this case the path "./template.html" would be relative to the base class, not the child class.

There are a few issues here:
- Ideally, the extension should just ignore any component it doesn't understand and continue.
- The underlying error (`TypeError: Cannot read property 'path' of undefined`) isn't logged and nor is the name of the problematic component or file, so it's hard to figure out what is wrong.
- I guess components inheriting the template from a base class is pretty unusual, but it would be good if this didn't break too much functionality and really nice if it was fully supported.

**Comment by ipatalas** [comment]
You're right. It shouldn't break other components. It was designed like this so that failing components shouldn't stop the extension to work. Something must have gone wrong then.

Regarding logging out should've been logged correctly. Don't know yet why it hasn't been but I'm gonna figure it out and fix hopefully.

It is pretty unusual I must say but also shouldn't be that hard to implement now. If it will be too much time consuming then I might go the easy way which is leave the component with only the parts from child class and ignore the fact it cannot read the rest. For sure it shouldn't stop the extension to work with other components.

**Comment by ipatalas** [comment]
There was one last place where the error wasn't logged - there are none now :) The error didn't happen during component parsing but rather when trying to show suggestions. That's why it broke the extension. That code is pretty reliable, I usually touch the parsing part, but still it can happen sometimes. At least it will now log the actual error in the console.
For now a component without a template will still be suggested in HTML files but the functionality will be limited. Due to the fact that the template is unknown Go To Definition will never show its template and also controller members auto-completion will not work since it won't know it's inside the template of this component.

I also did some research and implementing full support for such cases is not a quick task so I will have to postpone it for now. Splitting it to #17 so that it won't be forgotten and you can easily track it.

**Comment by stogle** [comment]
With this fix in 0.6.7, every component in my project is recognized! Thanks.
